benchmarks/scripts/send_usage_data_to_codespeed.py

```diff
@@ -299,25 +330,40 @@ def main(
     # Give it some time for the process to start and initialize before first capture
     time.sleep(5)
 
-    while time.time() <= end_time:
-        logger.debug("Capturing gauge metrics...")
-        capture_metrics(
-            tracker=tracker,
-            process=process,
-            metrics=METRICS_GAUGES,
-            values=captured_values,
-            capture_agent_status_metrics=capture_agent_status_metrics,
+    with open(BACKUP_PATH, "a") as backup:
+        current_time = time.time()
+        while current_time <= end_time:
+            current_time = time.time()
+            logger.debug("Capturing gauge metrics...")
+            captured = capture_metrics(
+                tracker=tracker,
+                process=process,
+                metrics=METRICS_GAUGES,
+                capture_agent_status_metrics=capture_agent_status_metrics,
+            )
+            # write captured metrics to backup file.
+            backup.write("{0}\n".format(json.dumps(captured)))
```

Comment:
Perhaps we should make this writing metrics to a file optional?

This way we can avoid additional interference / disk writes on non-kubernetes Circle CI benchmarks where everything runs inside a single container.

Reply:
EDIT: For simplicity I'm fine with keeping it as is (aka always write metrics to a file).

Writing every 10 seconds or so shouldn't add much overhead :)

Reply:
I refactored it a little bit and made file write optional. Hope it is easier to understand:)